projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -5619,10 +5647,14 @@ export class IgxGridBaseDirective extends DisplayDensityBase implements
         if (this.dataView.slice(rowIndex, rowIndex + 1).find(rec => rec.expression || rec.childGridsData)) {
             visibleColIndex = -1;
         }
-        const shouldScrollVertically = this.navigation.shouldPerformVerticalScroll(rowIndex, visibleColIndex);
+        // If the target row is pinned no need to scroll as well.
+        const shouldScrollVertically =
+            !this.isRecordPinnedByIndex(rowIndex) && this.navigation.shouldPerformVerticalScroll(rowIndex, visibleColIndex);
```

Comment:
As shouldPerformVerticalScroll is called in other places (mrl, hierarchical grid nav services) the check if a record is pinned should be moved inside the method to ensure consistency.